src/work-allocation/components/task-list-filter/task-list-filter.component.ts

```diff
@@ -251,8 +251,11 @@ export class TaskListFilterComponent implements OnInit, OnDestroy {
     };
     let baseLocation = null;
     // if there are no booking locations selected then check for base location for salary judge
-    if ((locations.length === 0) && this.route.snapshot.data && this.route.snapshot.data.locations) {
-      baseLocation = this.route.snapshot.data.locations;
+    if ((locations.length === 0) && this.route.snapshot.data && this.route.snapshot.data.location) {
+      const location: Location = this.route.snapshot.data.location;
+      if (location) {
+        baseLocation = [location];
+      }
```

Comment:
I believe the old code is what we want as the locations are being given in an array

Reply:
At least that is what Kartheek told me